doc/src/contributing/new-contributors-guide/workflow-process.md

```diff
@@ -0,0 +1,708 @@
+
+
+```{note}
+This guide is intended for people are already familiar with contributing to
+open source projects on GitHub. If you are new to GitHub, read the
+[](./dev-setup.md) guide first.
+```
+
+(workflow-process-checklist)=
+## Checklist for Contributions
+
+Here's a checklist of things that need to be done when making a pull request
+to SymPy. These things all need to be done before a pull request is merged. It
+is not necessary to do all of them before the pull request is opened at all,
+but it is often a good idea to check the basic things first before opening a
+pull request, or even before committing a change.
+
+- [ ] **Make sure [code quality
+  checks](workflow-process-code-quality) pass.**
+
+  ```bash
+  ./bin/test quality
+  flake8 sympy/
+  ```
+
+- [ ] **[Add tests](workflow-process-add-tests).** All new functionality
+  should be tested. Bug fixes should add regression tests. Tests are written
+  in pytest `assert f(x) == y` style and are included in corresponding `tests`
+  directories in the `sympy/` source. See the guide on [writing
+  tests](writing-tests).
+
+- [ ] **New public functions and methods should have a docstring.**
+
+- [ ] **Docstrings should include [doctests](writing-tests-doctests).**
+
+- [ ] **Make sure all tests pass.** [You may want to run a relevant subset of
+  the test suite locally](workflow-process-run-tests) before committing (e.g.,
+  `./bin/test solvers`). When you open a pull request, all tests will be run
+  on CI. The CI must be all green before a PR can be merged.
+
+- [ ] **[Write good commit messages](workflow-process-commit-messages).**
+
+- [ ] (First time contributors only) **[Add your name to the `.mailmap`
+  file](mailmap-instructions)**. The "test / authors" CI build on GitHub will
+  fail if this is not done correctly.
+
+- [ ] **Cross reference relevant issues in the pull request description.** If
+  the pull request fixes an issue (i.e., the issue should be closed once the
+  PR is merged), use the ["fixes #123" syntax
+  ](https://help.github.com/articles/closing-issues-via-commit-messages).
+
+- [ ] **Add a comment to the original issue cross-referencing the pull
+  request** for visibility. If there is not a corresponding issue, this is OK.
+  It is not necessary to open an issue unless there are further improvements
+  needed beyond your PR.
+
+- [ ] **[Add a release notes
+  entry](https://github.com/sympy/sympy/wiki/Writing-Release-Notes).** This
+  should be done when opening the pull request, in the pull request
+  description field. It can be edited at any time before the pull request is
+  merged.
+
+- [ ] **Respond to review comments.** All SymPy pull requests must be reviewed
+  by someone else before they can be merged.
+
+
+## Pick an issue to fix
+
+The best way to start with the main code base is to fix some existing bugs.
+Peruse the ["Easy to fix"
+issues](https://github.com/sympy/sympy/issues?q=is%3Aopen+is%3Aissue+label%3A%22Easy+to+Fix%22)
```

Comment:
The checklist above is good because it is nice and short for someone who already knows what they are doing in terms of git, github, python etc. That hypothetical person wants a quick list of what is needed that is partcular to sympy rather than say another project that they could contribute to via github. I think it is good to have a nice short page for that hypothetical contributor where it is assumed that the contributor already knows of an issue to be fixed and wants a low-friction interaction to submit a PR because they know how to do that.

From this point on though it moves into information that is not so relevant for someone who already knows what they are doing e.g. "don't commit to master" which I think does not belong in the same page.

Reply:
Of course ending this page with a bunch of links like "how to write tests", "how to build the docs" etc is good but this page should be kept short so that someone who knows what they are doing can read it entirely in a short time.

Reply:
Note that other places refer to this page as the one to read for someone who generally knows what they are doing so the hypothetical contributor has been directed here.

Reply:
Or maybe it's fine if the section above just ends with something like "that's all that is needed but below is some more information about various aspects of contributing and some links for quick navigation to the sections below.

Basically I want someone who knows how to improve the code and how to make a pull request not to have to waste a lot of time figuring out what to do. Asking that person to read long pages full of irrelevant stuff discourages contribution.

Reply:
A surprising number of people don't know that you shouldn't commit to master. There's actually quite a skill gap. In fact, if you polled SymPy contributors on how comfortable they are with git/GitHub I'd wager the results would skew toward low self-reported skill, even among people who actually do know a lot.

I guess the question here is whether this is better on this page or on a separate page. I don't really feel strongly either way. I ended up putting it on the same page because it felt better that way, as it makes it easier to cross-reference things on the same page. But it's really not a big deal. Hopefully it's clear that the checklist is just a summary of the sections below, but we can make it clearer if it isn't.

Reply:
Maybe there could just be some text that clarifies that the rest of the document does not need to be read for someone who clearly understands how to follow the steps in the checklist.

Reply:
Of course they can read it if they want but we make it clear that there are no unusual surprises down there for someone who knows what they are doing so they can skip it if it looks straight-forward.